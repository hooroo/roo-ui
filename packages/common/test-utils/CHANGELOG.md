# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
