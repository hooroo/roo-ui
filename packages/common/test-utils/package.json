{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
