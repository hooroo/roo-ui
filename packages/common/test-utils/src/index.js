export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
