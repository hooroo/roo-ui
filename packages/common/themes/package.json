{
  "name": "@roo-ui/themes",
  "version": "0.4.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
