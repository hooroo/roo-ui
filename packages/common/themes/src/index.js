import qantas from './qantas';

export { qantas }; // eslint-disable-line import/prefer-default-export
