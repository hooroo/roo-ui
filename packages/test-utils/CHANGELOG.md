# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.50.7"></a>
## [0.50.7](https://github.com/hooroo/roo-ui/compare/v0.50.6...v0.50.7) (2018-08-14)


### Bug Fixes

* pass theme to child components; update snapshots ([#153](https://github.com/hooroo/roo-ui/issues/153)) ([1a83477](https://github.com/hooroo/roo-ui/commit/1a83477))




<a name="0.39.1"></a>
## [0.39.1](https://github.com/hooroo/roo-ui/compare/v0.39.0...v0.39.1) (2018-06-26)




**Note:** Version bump only for package @roo-ui/test-utils

<a name="0.39.0"></a>
# [0.39.0](https://github.com/hooroo/roo-ui/compare/v0.38.4...v0.39.0) (2018-06-26)


### Features

* add build:watch scripts ([#98](https://github.com/hooroo/roo-ui/issues/98)) ([bcae0bf](https://github.com/hooroo/roo-ui/commit/bcae0bf))




<a name="0.32.0"></a>
# [0.32.0](https://github.com/hooroo/roo-ui/compare/v0.31.0...v0.32.0) (2018-06-15)


### Features

* add autocomplete component ([#77](https://github.com/hooroo/roo-ui/issues/77)) ([eee3f8c](https://github.com/hooroo/roo-ui/commit/eee3f8c))




<a name="0.19.0"></a>
# [0.19.0](https://github.com/hooroo/roo-ui/compare/v0.18.0...v0.19.0) (2018-05-22)


### Features

* Add <List/> & <ListItem> ([#49](https://github.com/hooroo/roo-ui/issues/49)) ([b922a8a](https://github.com/hooroo/roo-ui/commit/b922a8a))




<a name="0.7.0"></a>
# [0.7.0](https://github.com/hooroo/roo-ui/compare/v0.6.4...v0.7.0) (2018-05-11)


### Features

* add new icon set ([#14](https://github.com/hooroo/roo-ui/issues/14)) ([377c4c8](https://github.com/hooroo/roo-ui/commit/377c4c8))




<a name="0.5.4"></a>
## [0.5.4](https://github.com/hooroo/roo-ui/compare/v0.5.2...v0.5.4) (2018-05-08)


### Bug Fixes

* replace build script with prepublish ([90cafe3](https://github.com/hooroo/roo-ui/commit/90cafe3))




<a name="0.5.3"></a>
## [0.5.3](https://github.com/hooroo/roo-ui/compare/v0.5.2...v0.5.3) (2018-05-08)


### Bug Fixes

* replace build script with prepublish ([90cafe3](https://github.com/hooroo/roo-ui/commit/90cafe3))




<a name="0.5.0"></a>
# [0.5.0](https://github.com/hooroo/roo-ui/compare/v0.4.2...v0.5.0) (2018-05-07)


### Features

* add ui primitives ([#13](https://github.com/hooroo/roo-ui/issues/13)) ([5789b96](https://github.com/hooroo/roo-ui/commit/5789b96))




<a name="0.4.2"></a>
## [0.4.2](https://github.com/hooroo/roo-ui/compare/v0.4.0...v0.4.2) (2018-04-26)




**Note:** Version bump only for package @roo-ui/test-utils

<a name="0.4.1"></a>
## [0.4.1](https://github.com/hooroo/roo-ui/compare/v0.4.0...v0.4.1) (2018-04-26)




**Note:** Version bump only for package @roo-ui/test-utils

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
