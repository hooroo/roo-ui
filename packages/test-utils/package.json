{
  "name": "@roo-ui/test-utils",
  "version": "0.5.3",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "prepublish": "babel src -d dist --copy-files"
  }
}
