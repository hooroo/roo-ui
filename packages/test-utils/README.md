# test-utils

## Installation

```shell
$ yarn add @roo-ui/test-utils
```

## Usage

### `shallowWithTheme`

Shallow render (with [Enzyme](http://airbnb.io/enzyme/)) a component along with theme context.

```js
import { shallowWithTheme } from '@roo-ui/test-utils';
import { qantas as theme } from '@roo-ui/themes';

import Example from '.';

describe('<Example />', () => {
  it('renders correctly', () => {
    expect(shallowWithTheme(<Example />, theme)).toMatchSnapshot;
  });
});
```
