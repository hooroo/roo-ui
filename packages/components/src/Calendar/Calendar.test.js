import React from 'react';
import { qantas as theme } from '@roo-ui/themes';
import { shallowWithTheme } from '@roo-ui/test-utils';

import { Calendar } from '.';

describe('<Calendar />', () => {
  let wrapper;
  let childrenWrapper;

  const props = {
    date: new Date('2018-07-03'),
    minDate: new Date('2018-06-01'),
    maxDate: new Date('2018-08-20'),
    selected: new Date('2018-07-20'),
    onDateSelected: jest.fn(),
    monthsToDisplay: 1,
    stacked: true,
    weekdayNames: ['S', 'M', 'T', 'W', 'T', 'F', 'S'],
    blockedDates: [new Date('2018-07-04'), new Date('2018-07-05')],
  };

  beforeEach(() => {
    wrapper = shallowWithTheme(<Calendar {...props} />, theme);
    childrenWrapper = wrapper.dive();
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    expect(childrenWrapper).toMatchSnapshot();
  });

  describe('<Dayzed />', () => {
    it('passes down props.monthsToDisplay', () => {
      expect(wrapper.find('Dayzed').prop('monthsToDisplay')).toEqual(props.monthsToDisplay);
    });

    it('assigns the rest of the props', () => {
      expect(wrapper.props()).toEqual(expect.objectContaining({
        date: props.date,
        minDate: props.minDate,
        maxDate: props.maxDate,
        selected: props.selected,
        onDateSelected: props.onDateSelected,
      }));
    });
  });

  describe('<CalendarMonth />', () => {
    it('passes the month name to each calendar month', () => {
      expect(childrenWrapper.find('CalendarMonth').prop('month')).toEqual('Jul');
    });

    it('passes down props.stacked', () => {
      expect(childrenWrapper.find('CalendarMonth').prop('stacked')).toEqual(props.stacked);
    });

    it('passes down props.monthsToDisplay', () => {
      expect(childrenWrapper.find('CalendarMonth').prop('monthsToDisplay')).toEqual(props.monthsToDisplay);
    });
  });

  describe('<CalendarWeekday />', () => {
    it('renders one for each weekday name', () => {
      expect(childrenWrapper.find('CalendarWeekday')).toHaveLength(props.weekdayNames.length);
    });
  });

  describe('<CalendarEmptyDay />', () => {
    it('renders one for each empty day in the calendar month', () => {
      expect(childrenWrapper.find('CalendarEmptyDay')).toHaveLength(11);
    });
  });

  describe('<CalendarDay />', () => {
    it('it renders a day for each day in the month', () => {
      expect(childrenWrapper.find('CalendarDay')).toHaveLength(31);
    });

    it('renders days as clickable elements', () => {
      const day = childrenWrapper.find('CalendarDay').first();

      expect(day.props()).toEqual(expect.objectContaining({
        selectable: true,
        disabled: false,
      }));
    });

    it('adds props.selected true when date is in props.selected', () => {
      const day19 = childrenWrapper.find('CalendarDay').at(19);

      expect(day19.props()).toEqual(expect.objectContaining({
        selected: true,
      }));
    });

    it('renders a blocked day when date is in props.blockedDates', () => {
      const day3 = childrenWrapper.find('CalendarDay').at(3);
      const day4 = childrenWrapper.find('CalendarDay').at(4);

      expect(day3.props()).toEqual(expect.objectContaining({
        selected: false,
        selectable: false,
        disabled: false,
      }));

      expect(day4.props()).toEqual(expect.objectContaining({
        selected: false,
        selectable: false,
        disabled: false,
      }));
    });
  });
});
