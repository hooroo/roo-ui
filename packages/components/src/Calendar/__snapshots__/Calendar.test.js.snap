// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> renders children correctly 1`] = `
<Box>
  <CalendarNav
    nextProps={
      Object {
        "aria-label": "Go forward 1 month",
        "disabled": false,
        "onClick": [Function],
      }
    }
    prevProps={
      Object {
        "aria-label": "Go back 1 month",
        "disabled": false,
        "onClick": [Function],
      }
    }
  />
  <Box
    flexWrap="wrap"
  >
    <CalendarMonth
      key="62018"
      month="Jul"
      monthsToDisplay={1}
      year={2018}
    >
      <Box>
        <CalendarWeekday
          key="62018S"
        >
          S
        </CalendarWeekday>
        <CalendarWeekday
          key="62018M"
        >
          M
        </CalendarWeekday>
        <CalendarWeekday
          key="62018T"
        >
          T
        </CalendarWeekday>
        <CalendarWeekday
          key="62018W"
        >
          W
        </CalendarWeekday>
        <CalendarWeekday
          key="62018T"
        >
          T
        </CalendarWeekday>
        <CalendarWeekday
          key="62018F"
        >
          F
        </CalendarWeekday>
        <CalendarWeekday
          key="62018S"
        >
          S
        </CalendarWeekday>
      </Box>
      <Box>
        <CalendarDay
          aria-label="Sun Jul 01 2018"
          aria-pressed={false}
          disabled={false}
          key="201860"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          1
        </CalendarDay>
        <CalendarDay
          aria-label="Mon Jul 02 2018"
          aria-pressed={false}
          disabled={false}
          key="201861"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          2
        </CalendarDay>
        <CalendarDay
          aria-label="Tue Jul 03 2018"
          aria-pressed={true}
          disabled={false}
          key="201862"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={true}
        >
          3
        </CalendarDay>
        <CalendarDay
          aria-label="Wed Jul 04 2018"
          aria-pressed={false}
          disabled={false}
          key="201863"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          4
        </CalendarDay>
        <CalendarDay
          aria-label="Thu Jul 05 2018"
          aria-pressed={false}
          disabled={false}
          key="201864"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          5
        </CalendarDay>
        <CalendarDay
          aria-label="Fri Jul 06 2018"
          aria-pressed={false}
          disabled={false}
          key="201865"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          6
        </CalendarDay>
        <CalendarDay
          aria-label="Sat Jul 07 2018"
          aria-pressed={false}
          disabled={false}
          key="201866"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          7
        </CalendarDay>
        <CalendarDay
          aria-label="Sun Jul 08 2018"
          aria-pressed={false}
          disabled={false}
          key="201860"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          8
        </CalendarDay>
        <CalendarDay
          aria-label="Mon Jul 09 2018"
          aria-pressed={false}
          disabled={false}
          key="201861"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          9
        </CalendarDay>
        <CalendarDay
          aria-label="Tue Jul 10 2018"
          aria-pressed={false}
          disabled={false}
          key="201862"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          10
        </CalendarDay>
        <CalendarDay
          aria-label="Wed Jul 11 2018"
          aria-pressed={false}
          disabled={false}
          key="201863"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          11
        </CalendarDay>
        <CalendarDay
          aria-label="Thu Jul 12 2018"
          aria-pressed={false}
          disabled={false}
          key="201864"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          12
        </CalendarDay>
        <CalendarDay
          aria-label="Fri Jul 13 2018"
          aria-pressed={false}
          disabled={false}
          key="201865"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          13
        </CalendarDay>
        <CalendarDay
          aria-label="Sat Jul 14 2018"
          aria-pressed={false}
          disabled={false}
          key="201866"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          14
        </CalendarDay>
        <CalendarDay
          aria-label="Sun Jul 15 2018"
          aria-pressed={false}
          disabled={false}
          key="201860"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          15
        </CalendarDay>
        <CalendarDay
          aria-label="Mon Jul 16 2018"
          aria-pressed={false}
          disabled={false}
          key="201861"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          16
        </CalendarDay>
        <CalendarDay
          aria-label="Tue Jul 17 2018"
          aria-pressed={false}
          disabled={false}
          key="201862"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          17
        </CalendarDay>
        <CalendarDay
          aria-label="Wed Jul 18 2018"
          aria-pressed={false}
          disabled={false}
          key="201863"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          18
        </CalendarDay>
        <CalendarDay
          aria-label="Thu Jul 19 2018"
          aria-pressed={false}
          disabled={false}
          key="201864"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          19
        </CalendarDay>
        <CalendarDay
          aria-label="Fri Jul 20 2018"
          aria-pressed={false}
          disabled={false}
          key="201865"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          20
        </CalendarDay>
        <CalendarDay
          aria-label="Sat Jul 21 2018"
          aria-pressed={false}
          disabled={false}
          key="201866"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          21
        </CalendarDay>
        <CalendarDay
          aria-label="Sun Jul 22 2018"
          aria-pressed={false}
          disabled={false}
          key="201860"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          22
        </CalendarDay>
        <CalendarDay
          aria-label="Mon Jul 23 2018"
          aria-pressed={false}
          disabled={false}
          key="201861"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          23
        </CalendarDay>
        <CalendarDay
          aria-label="Tue Jul 24 2018"
          aria-pressed={false}
          disabled={false}
          key="201862"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          24
        </CalendarDay>
        <CalendarDay
          aria-label="Wed Jul 25 2018"
          aria-pressed={false}
          disabled={false}
          key="201863"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          25
        </CalendarDay>
        <CalendarDay
          aria-label="Thu Jul 26 2018"
          aria-pressed={false}
          disabled={false}
          key="201864"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          26
        </CalendarDay>
        <CalendarDay
          aria-label="Fri Jul 27 2018"
          aria-pressed={false}
          disabled={false}
          key="201865"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          27
        </CalendarDay>
        <CalendarDay
          aria-label="Sat Jul 28 2018"
          aria-pressed={false}
          disabled={false}
          key="201866"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          28
        </CalendarDay>
        <CalendarDay
          aria-label="Sun Jul 29 2018"
          aria-pressed={false}
          disabled={false}
          key="201860"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          29
        </CalendarDay>
        <CalendarDay
          aria-label="Mon Jul 30 2018"
          aria-pressed={false}
          disabled={false}
          key="201861"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          30
        </CalendarDay>
        <CalendarDay
          aria-label="Tue Jul 31 2018"
          aria-pressed={false}
          disabled={false}
          key="201862"
          onClick={[Function]}
          role="button"
          selectable={true}
          selected={false}
        >
          31
        </CalendarDay>
        <Box
          key="201863"
        />
        <Box
          key="201864"
        />
        <Box
          key="201865"
        />
        <Box
          key="201866"
        />
      </Box>
    </CalendarMonth>
  </Box>
</Box>
`;

exports[`<Calendar /> renders correctly 1`] = `
<Dayzed
  date={2018-07-03T00:00:00.000Z}
  firstDayOfWeek={0}
  maxDate={2018-08-20T00:00:00.000Z}
  minDate={2018-06-01T00:00:00.000Z}
  monthsToDisplay={1}
  onDateSelected={[MockFunction]}
  onOffsetChanged={[Function]}
  render={[Function]}
  selected={2018-07-03T00:00:00.000Z}
  showOutsideDays={false}
/>
`;
